compiler/plugins/target/AMD-AIE/aie/AMDAIEDmaToNpu.cpp

```diff
@@ -359,25 +359,25 @@ struct DmaToNpuPattern : OpConversionPattern<NpuDmaMemcpyNdOp> {
     buffer_offset = IntegerAttr::get(i32ty, offset);
 
     // d0_size
-    if (strides[0]) d0_size = IntegerAttr::get(i32ty, sizes[0]);
+    if (strides[1]) d0_size = IntegerAttr::get(i32ty, sizes[0]);
 
     // d0_stride
-    d0_stride = IntegerAttr::get(i32ty, 0);
+    if (strides[0]) d0_stride = IntegerAttr::get(i32ty, strides[0] - 1);
 
     // d1_size
-    if (strides[1]) d1_size = IntegerAttr::get(i32ty, sizes[1]);
+    if (strides[2]) d1_size = IntegerAttr::get(i32ty, sizes[1]);
 
     // d1_stride
-    if (strides[0]) d1_stride = IntegerAttr::get(i32ty, strides[0] - 1);
+    if (strides[1]) d1_stride = IntegerAttr::get(i32ty, strides[1] - 1);
 
     // d2_stride
-    if (strides[1]) d2_stride = IntegerAttr::get(i32ty, strides[1] - 1);
+    if (strides[2]) d2_stride = IntegerAttr::get(i32ty, strides[2] - 1);
 
     // iteration_size
-    if (strides[2]) iteration_size = IntegerAttr::get(i32ty, sizes[3] - 1);
+    if (strides[3]) iteration_size = IntegerAttr::get(i32ty, sizes[3] - 1);
 
     // iteration_stride
-    if (strides[2]) iteration_stride = IntegerAttr::get(i32ty, strides[2] - 1);
+    if (strides[3]) iteration_stride = IntegerAttr::get(i32ty, strides[3] - 1);
```

Comment:
thanks @erwei-xilinx - I saw that PR and made a mental note to bring it over but then got distracted.